crates/uv/tests/it/sync.rs

```diff
@@ -1782,6 +1782,223 @@ fn sync_non_existent_group() -> Result<()> {
+
+    Usage: uv sync --cache-dir [CACHE_DIR] --only-dev --exclude-newer <EXCLUDE_NEWER>
+
+    For more information, try '--help'.
+    ");
+
+    // --all-groups and --only-dev should error
+    uv_snapshot!(context.filters(), context.sync()
+        .arg("--all-groups")
+        .arg("--only-dev"), @r"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: the argument '--all-groups' cannot be used with '--only-dev'
+
+    Usage: uv sync --cache-dir [CACHE_DIR] --all-groups --exclude-newer <EXCLUDE_NEWER>
+
+    For more information, try '--help'.
+    ");
+
+    // --all-groups and --only-group should error
+    uv_snapshot!(context.filters(), context.sync()
+        .arg("--all-groups")
+        .arg("--only-group").arg("bar"), @r"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: the argument '--all-groups' cannot be used with '--only-group <ONLY_GROUP>'
+
+    Usage: uv sync --cache-dir [CACHE_DIR] --all-groups --exclude-newer <EXCLUDE_NEWER>
+
+    For more information, try '--help'.
+    ");
+
+    // --group and --only-group should error if they name disjoint things
+    uv_snapshot!(context.filters(), context.sync()
+        .arg("--group").arg("foo")
+        .arg("--only-group").arg("bar"), @r"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: the argument '--group <GROUP>' cannot be used with '--only-group <ONLY_GROUP>'
+
+    Usage: uv sync --cache-dir [CACHE_DIR] --group <GROUP> --exclude-newer <EXCLUDE_NEWER>
+
+    For more information, try '--help'.
+    ");
+
+    // --group and --only-group should error if they name same things
+    // (This one would be fair to allow, but... is it worth it?)
+    uv_snapshot!(context.filters(), context.sync()
+        .arg("--group").arg("foo")
+        .arg("--only-group").arg("foo"), @r"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: the argument '--group <GROUP>' cannot be used with '--only-group <ONLY_GROUP>'
+
+    Usage: uv sync --cache-dir [CACHE_DIR] --group <GROUP> --exclude-newer <EXCLUDE_NEWER>
+
+    For more information, try '--help'.
+    ");
+
+    // --all-groups and --no-default-groups is redundant but should be --all-groups
+    uv_snapshot!(context.filters(), context.sync()
+        .arg("--all-groups")
+        .arg("--no-default-groups"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 9 packages in [TIME]
+    Prepared 8 packages in [TIME]
+    Installed 8 packages in [TIME]
+     + certifi==2024.2.2
+     + charset-normalizer==3.3.2
+     + idna==3.6
+     + iniconfig==2.0.0
+     + requests==2.31.0
+     + sniffio==1.3.1
+     + typing-extensions==4.10.0
+     + urllib3==2.2.1
+    ");
+
+    // --dev --only-dev should saturate as --only-dev
+    uv_snapshot!(context.filters(), context.sync()
```

Comment:
Weird to allow this but not `// --group and --only-group should error if they name same things` but... eh.

Reply:
Preserving previous behaviour :)